backend/src/auth/repositories/auth_repo.py

```diff
@@ -1,32 +1,62 @@
-from fastapi import HTTPException
+import logging
+import uuid
+from datetime import timedelta
+from typing import Optional
+
+from fastapi import Response
+from sqlalchemy.exc import SQLAlchemyError
 from sqlalchemy.orm import Session
 
-from src.auth.interfaces.repository import Repository
+from src.auth.interfaces import Repository
 from src.auth.models import User
-from src.auth.schemas import SignInSchema, UserSchema
+from src.auth.schemas import (
+    RegisterRequest,
+    RegisterResponse,
+    SignInRequest,
+    SignInResponse,
+    UserResponse,
+)
 from src.auth.utils import check_password, hash_password, return_token
 
+logger = logging.getLogger(__name__)
+
 
 class AuthRepository(Repository):
-    def register(self, user: UserSchema, db: Session) -> dict:
+    def get_user_by_id(self, user_id: uuid.UUID, db: Session) -> Optional[UserResponse]:
+        try:
+            user = db.query(User).filter_by(id=user_id).first()
+            return user
+        except Exception as e:
+            logger.error(f"Failed to get user: {e}")
+
+    def get_user(self, username: str, db: Session) -> Optional[UserResponse]:
+        try:
+            user = db.query(User).filter_by(username=username).first()
+            return user
+        except Exception as e:
+            logger.error(f"Failed to get user: {e}")
+
+    def authenticate(self, username: str, password: str, db: Session) -> bool:
+        user = self.get_user(username, db)
+        if not user:
+            return False
+        if not check_password(password, user.password):
+            return False
+        return True
```

Comment:
What is the meaning of this method ? WHy are you checking passwords in persistance layer. This method is wrong and shouldnt be here.

Reply:
btw you can just 
return check_password(password, user.password)

Reply:
I still haven't gotten used to it